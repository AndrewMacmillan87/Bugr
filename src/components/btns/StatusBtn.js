import React from "react";

import PropTypes from "prop-types";

import "./status-btn.css";

const StatusBtn = ({ id, value }) => {
    return (
        <button className='issue-state-btn' id={id} type='button'>
            {value}
        </button>
    );
};

StatusBtn.propTypes = {
    id: PropTypes.string,
    value: PropTypes.string,
};

export default StatusBtn;
