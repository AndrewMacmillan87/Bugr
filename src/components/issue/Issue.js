import React, { useState } from "react";

import StatusBtn from "../btns/StatusBtn";
import StatusSelector from "../status_selector/StatusSelector";

import PropTypes from "prop-types";

import "./issue.css";

export const Issue = ({ id, description, completion, priority, stage }) => {
    const [showPriorities, setPriorities] = useState(false);
    const [showStages, setStages] = useState(false);
    const [showIssueOptions, setIssueOptions] = useState(false);
    const [showDateOptions, setDateOptions] = useState(false);

    const stageId = `stage-${id}`;
    const priorityId = `priority-${id}`;

    return (
        <div className='issue-container'>
            <div className='issue-description-container'>
                <div className='issue-description'>{description}</div>
                <i
                    className='issue-menu fas fa-ellipsis-v'
                    title='Options'
                    onClick={() => setIssueOptions(true)}></i>
                <div className='issue-options-temp-container'>
                    {showIssueOptions && (
                        <div className='issue-options-container'>
                            <i
                                className='remove-menu-options fas fa-times-circle'
                                onClick={() => setIssueOptions(false)}></i>
                            <ul className='issue-options-list'>
                                <li className='issue-option-list-element'>
                                    New issue
                                </li>
                                <li className='issue-option-list-element'>
                                    Edit
                                </li>
                                <li className='issue-option-list-element'>
                                    Delete
                                </li>
                                <li className='issue-option-list-element'>
                                    Notes
                                </li>
                            </ul>
                        </div>
                    )}
                </div>
            </div>
            <div className='issue-state-btn-container'>
                {showStages && (
                    <StatusSelector status='stage' onClick={setStages} />
                )}
                <StatusBtn
                    id={stageId}
                    value={stage}
                    onClick={() => {
                        setStages(true);
                        setPriorities(false);
                    }}
                />
                {showPriorities && (
                    <StatusSelector status='priority' onClick={setPriorities} />
                )}
                <StatusBtn
                    id={priorityId}
                    value={priority}
                    onClick={() => {
                        setPriorities(true);
                        setStages(false);
                    }}
                />
            </div>
            <div className='date-container'>
                <div className='completion-date'>{completion}</div>
                <i
                    className='issue-menu fas fa-ellipsis-v'
                    title='Date options'
                    onClick={() => setDateOptions(true)}></i>
                <div className='date-options-temp-container'>
                    {showDateOptions && (
                        <div className='date-options-container'>
                            <i
                                className='remove-menu-options fas fa-times-circle'
                                onClick={() => setDateOptions(false)}></i>
                            <ul className='date-options-list'>
                                <li className='date-option-list-element'>
                                    Edit
                                </li>
                                <li className='date-option-list-element'>
                                    Delete
                                </li>
                            </ul>
                        </div>
                    )}
                </div>
            </div>
        </div>
    );
};

Issue.propTypes = {
    id: PropTypes.string,
    description: PropTypes.string,
    completion: PropTypes.string,
    priority: PropTypes.string,
    stage: PropTypes.string,
};

export default Issue;
