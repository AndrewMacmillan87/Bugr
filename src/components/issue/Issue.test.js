import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import Issue from "./Issue";
import StatusBtn from "../btns/StatusBtn";

describe("Issue component", () => {
    test("Renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<Issue />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("Renders correctly", () => {
        const tree = renderer.create(<Issue />);
        expect(tree).toMatchSnapshot();
    });

    describe("Issue HTML elements", () => {
        it("should render 6 div elements", () => {
            const wrapper = mount(<Issue />);
            expect(wrapper.find("div").length).toEqual(6);
        });

        it("should render 2 i elements", () => {
            const wrapper = mount(<Issue />);
            expect(wrapper.find("i").length).toEqual(2);
        });

        it("should render 2 StatusBtn components", () => {
            const wrapper = mount(<Issue />);
            expect(wrapper.find(StatusBtn).length).toEqual(2);
        });
    });
});
