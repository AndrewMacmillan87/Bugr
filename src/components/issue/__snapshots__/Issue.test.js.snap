// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue component Renders correctly 1`] = `
<div
  className="issue-container"
>
  <div
    className="issue-description-container"
  >
    <div
      className="issue-description"
    />
    <i
      className="issue-menu fas fa-ellipsis-v"
    />
  </div>
  <div
    className="issue-state-btn-container"
  >
    <button
      className="issue-state-btn"
      id="priority"
      type="button"
    />
    <button
      className="issue-state-btn"
      id="stage"
      type="button"
    />
  </div>
  <div
    className="date-container"
  >
    <div
      className="completion-date"
    />
    <i
      className="issue-menu fas fa-ellipsis-v"
    />
  </div>
</div>
`;
